sale_purchase_requisition/models/sale_order.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    purchase_agent = fields.Many2one(
+        comodel_name="res.users", string="Comercial compra"
+    )
+
+    purchase_requisition_ids = fields.One2many(
+        comodel_name="purchase.requisition",
+        inverse_name="sale_order_id",
+        string="Acuerdo de compra",
+    )
+
+    def action_view_sale_requisition(self):
+        return {
+            "name": _("Purchase Requisition"),
+            "type": "ir.actions.act_window",
+            "res_model": "purchase.requisition",
+            "view_mode": "form",
+            "res_id": self.purchase_requisition_ids.id,
+        }
+
+    def _create_lines_preq(self):
+        self.ensure_one()
+        pur_req_lines = []
+        for line in self.order_line:
+            if line.product_id.purchase_ok:
+                pur_req_lines.append(
+                    [
+                        0,
+                        0,
+                        {
+                            "product_id": line.product_id.id,
+                            "product_qty": line.product_qty,
+                            "product_uom_id": line.product_uom.id,
```

Comment:
Price unit is not propagated. Is it intended?

```suggestion
                            "product_uom_id": line.product_uom.id,
                            "price_unit": line.price_unit,
```